Dockerfile

```diff
@@ -6,20 +6,23 @@ RUN mkdir -p /usr/src/app
 RUN mkdir /usr/src/app/private
 WORKDIR /usr/src/app
 
-ARG NODE_ENV
-ENV NODE_ENV $NODE_ENV
-
 COPY package.json package-lock.json /usr/src/app/
 # Without the gh-badges package.json and CLI script in place, `npm ci` will fail.
 COPY gh-badges /usr/src/app/gh-badges/
-RUN npm ci
+
+# We need dev deps to build the front end.
+RUN NODE_ENV=development npm ci
 
 COPY . /usr/src/app
 RUN npm run build
 RUN npm prune --production
 RUN npm cache clean --force
 
+# Run the server using production configs.
+ENV NODE_ENV production
+
 # Do we need to list the environment variables here?
```

Comment:
@calebcartwright Does that mean we don't need to explicitly list any env vars here?

Reply:
Just reviewing all the env vars I'm setting on my `docker run` command, other than the tokens/secrets for various badges, and proxy config that's specific to where I'm running, I believe the only relevant one is the `-e BIND_ADDRESS=0.0.0.0` 

I'm also still adding `-e PORT=80 -e BASE_URL=http://localhost:8080` but I don't think either of those are necessary

Reply:
`-e BASE_URL=http://localhost:8080` is also likely to not work when you run the container in your production environment, whereas there's a decent chance nothing will work. 😁

Reply:
Oddly enough, the app runs fine in the container, even with that arg, but I probably need to clean some of this up. 

I got it working a few months ago and was happy to leave it untouched as long as it kept running smoothly

Reply:
Including the frontend?

I'd imagine if the frontend were used on another machine, it would generate badge URLs that include `localhost`.

Reply:
Yup! I don't we actually use that env var anywhere do we?

Reply:
Oh! Ha! Yes we do, but only at build time!